bibliopixel/drivers/SPI/APA102.py

```diff
@@ -1,24 +1,43 @@
 from ... import gamma
 from .. channel_order import ChannelOrder
+from ... import data_maker
 from . base import SPIBase
 
 
 class APA102(SPIBase):
-    """Driver for APA102 based LED strips on devices like the Raspberry Pi and BeagleBone"""
+    """Driver for APA102/SK9822 based LED strips on devices like the Raspberry Pi and BeagleBone"""
 
     def __init__(self, num, gamma=gamma.APA102, **kwargs):
         super().__init__(num, gamma=gamma, **kwargs)
+        maker = kwargs.get('maker', data_maker.MAKER)
 
-        # APA102 requires latch bytes at the end
-        self._latch_bytes = [0xFF, 0, 0, 0] * ((num // 64) + 1)
+        # APA102/SK9822 requires latch bytes at the end)
+        # Many thanks to this article for combined APA102/SK9822 protocol
+        # https://cpldcpu.com/2016/12/13/sk9822-a-clone-of-the-apa102/
+        self._start_frame = 4  # start frame is [0, 0, 0, 0]
+        self._pixel_bytes = self.bufByteCount() + self.numLEDs  # 1 extra for brightness
```

Comment:
Ohh.... ha, yeah. That's kind of a cheat... bufByteCount() is (numLEDs * 3) - There was plans to maybe support layouts with different numbers of colors but that really never happened - But the APA102 has a 4 byte frame, so: (numLEDs * 3) + numLEDs = numLEDs * 4.
That's old code. Was trying to respect the bufByteCount thing. Which doesn't really wake a lot of sense. Will change.